unlock-app/src/middlewares/walletMiddleware.js

```diff
@@ -125,15 +125,29 @@ export default function walletMiddleware({ getState, dispatch }) {
         ensureReadyBefore(() => {
           walletService.createLock(action.lock, getState().account.address)
           if (config.services.storage) {
-            generateJWTToken(walletService, getState().account.address, {
-              lock: action.lock,
-            })
+            generateSignature(
+              walletService.web3,
+              getState().account.address,
```

Comment:
Didn't catch this before, but `account` can be `null`. You'll want to bail out at the top of `ensureReadyBefore` with false if `getState().account` is falsy

Reply:
@cellog Do you now if the `walletService.ready` check inside of the declaration of `ensureReadyBefore` would handle the general case of an account not being available

Reply:
probably. I'll leave this as resolved because of that.